src/Storages/StorageMergeTree.cpp

```diff
@@ -2178,6 +2178,9 @@ void StorageMergeTree::movePartitionToTable(const StoragePtr & dest_table, const
                         this->getStoragePolicy()->getName(), dest_table_storage->getStorageID().getNameForLogs(),
                         dest_table_storage->getStoragePolicy()->getName());
 
+    // Use the same back-pressure (delay/throw) logic as for INSERTs to be consistent and avoid possibility of exceeding part limits using MOVE PARTITION queries
+    dest_table_storage->delayInsertOrThrowIfNeeded(nullptr, local_context, true);
```

Comment:
Is it Ok to do this while holding locks and blocking merges? https://github.com/ClickHouse/ClickHouse/pull/62420/files#diff-205c1b8d661c01d47b9831ecd03be2e25db3a49192f186a00c8896e13eac77efR2164-R2166

Maybe it's better to move the delay before locks?